onnx_tf/handlers/backend/onehot.py

```diff
@@ -2,37 +2,70 @@
 import tensorflow as tf
 
 from onnx_tf.common import exception
+from onnx_tf.common import data_type
+from onnx_tf.common import sys_config
+from onnx_tf.common.tf_helper import tf_shape
 from onnx_tf.handlers.backend_handler import BackendHandler
 from onnx_tf.handlers.handler import onnx_op
-from onnx_tf.handlers.handler import partial_support
-from onnx_tf.handlers.handler import ps_description
 from onnx_tf.handlers.handler import tf_func
 
 
 @onnx_op("OneHot")
 @tf_func(tf.one_hot)
-@partial_support(True)
-@ps_description("OneHot indices in uint16/uint32/uint64/int8/int16/"+
-                "float16/float/double, or " +
-                "OneHot depth in uint8/uint16/uint32/uint64/int8/" +
-                "int16/int64/float16/float/double " +
-                "are not supported in Tensorflow.")
 class OneHot(BackendHandler):
-
+  indices_supported_type = [tf.uint8, tf.int32, tf.int64]
+  indices_cast_map = {
+      tf.uint16: tf.int32,
+      tf.uint32: tf.int64,
+      tf.int8: tf.int32,
+      tf.int16: tf.int32,
+      # ONNX spec state that all non-integer type will be casted to int64 before use
+      tf.float16: tf.int64,
+      tf.float32: tf.int64,
+      tf.float64: tf.int64
```

Comment:
Should this be handled similarly to the way depth is handled? That means we always check auto-cast for down-cast.